gluten-arrow/src/main/java/org/apache/gluten/init/NativeBackendInitializer.java

```diff
@@ -17,22 +17,33 @@
 package org.apache.gluten.init;
 
 import org.apache.gluten.GlutenConfig;
-import org.apache.gluten.backend.Backend;
 import org.apache.gluten.utils.ConfigUtil;
 
 import org.apache.spark.util.SparkShutdownManagerUtil;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
 import java.util.concurrent.atomic.AtomicBoolean;
 
 import scala.runtime.BoxedUnit;
 
 // Initialize native backend before calling any native methods from Java side.
 public final class NativeBackendInitializer {
   private static final Logger LOG = LoggerFactory.getLogger(NativeBackendInitializer.class);
-  private static final AtomicBoolean initialized = new AtomicBoolean(false);
+  private static final Map<String, NativeBackendInitializer> instances = new ConcurrentHashMap<>();
```

Comment:
Since `This doesn't tend to make Velox backend and CH backend work together`, why we need a backend map here?

Reply:
This change is not tended actually. It's just because the module `gluten-arrow` can no longer get a precise backend name because it's decoupled with `gluten-substrait` so it may see multiple backends loaded. There is another option to move this class to `backends-velox` so no such change is needed.